### PR TITLE
Add support for periodic tasks with cron schedules

### PR DESCRIPTION
**Pull request by aduane**
This would be paired with a recurring task in home-app that just calls send_to_plugin_runner every minute with the timestamp. (I have that code working and will make a PR tomorrow.)

This PR also makes the distinction between a protocol and other, non-protocol event handlers. A CronTask is not a protocol. It is an event handler that responds to the CRON event, and expects a SCHEDULE constant and an execute method to be defined.

### CODE REVIEW

File: canvas_sdk/handlers/cron_task.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class CronTask(BaseHandler):
+    """
+    A type of handler that executes periodically according to a provided schedule.
+    """
+
+    RESPONDS_TO = EventType.Name(EventType.CRON)
+
+    # Every minute unless specified
+    SCHEDULE: str = "* * * * *"
```
Comment (aduane) [inline canvas_sdk/handlers/cron_task.py:19]: Unsure if we should set a default like this.

File: canvas_sdk/handlers/cron_task.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        """
+        Perform some work and return a list of effects.
+        """
+
+    def compute(self) -> list[Effect]:
+        """
+        See if the task should execute given the timestamp
+        """
+        datetime = arrow.get(self.target).datetime
+        if datetime in Cron(self.SCHEDULE):
```
Comment (aduane) [inline canvas_sdk/handlers/cron_task.py:32]: Returns True if the given datetime is part of the cron schedule.

File: plugin_runner/sandbox.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
 import ast
 import builtins
-
+from _ast import AnnAssign
```
Comment (aduane) [inline plugin_runner/sandbox.py:3]: pre-commit sorted this.

**Comment by aduane** [comment]
I love the simplicity of this.

**Review by csande — APPROVED** [review]

**Pull request merged**